gpytorch/models/exact_prediction_strategies.py

```diff
@@ -40,7 +40,17 @@ def __init__(self, train_inputs, train_prior_dist, train_labels, likelihood, roo
         self._train_shape = train_prior_dist.event_shape
 
         # Flatten the training labels
-        train_labels = train_labels.reshape(*train_labels.shape[: -len(self.train_shape)], self._train_shape.numel())
+        try:
+            train_labels = train_labels.reshape(
+                *train_labels.shape[: -len(self.train_shape)], self._train_shape.numel()
+            )
+        except RuntimeError:
+            raise RuntimeError(
+                "Flattening the training labels failed. The most common cause of this error "
+                + "is that the shapes of the prior mean and the training labels are mismatched."
+                + "The shape of the train targets is {0}, ".format(train_labels.shape)
```

Comment:
Looks like there was still a space missing here.

```suggestion
                "Flattening the training labels failed. The most common cause of this error is "
                + "that the shapes of the prior mean and the training labels are mismatched. "
                + "The shape of the train targets is {0}, ".format(train_labels.shape)
```